pest/src/stack.rs

```diff
@@ -54,6 +54,13 @@ impl<T: Clone> Stack<T> {
         popped
     }
 
+    /// Dump the full state of the stack
+    pub fn get_state(&self) -> Vec<T> {
+        let mut stack = self.cache.clone();
```

Comment:
Calling this too often is expensive. Why not return a `&Vec<T>` here and `iter().rev()` later?